asteroid/engine/schedulers.py

```diff
@@ -1,9 +1,11 @@
 from torch.optim.optimizer import Optimizer
 
 
-class _BaseScheduler(Optimizer):
+class _BaseScheduler(object):
     def __init__(self, optimizer):
-        super(_BaseScheduler, self).__init__(optimizer.__dict__["param_groups"], optimizer.__dict__["defaults"])
+        self.__class__ = type(optimizer.__class__.__name__,
```

Comment:
I wonder if this will make problems with saving/loading models. I guess `_BaseScheduler` instances can't be serialized (e.g. pickled) because of this dynamic class creation.

I guess this is a workaround for a limitation in lightning; wouldn't it be better to fix it in lightning?

Reply:
yes exactly... it fails with distributed data parallel because it uses multiprocessing and it requires instances to be serialized. 
Works fine however with no ddp or dp.. it is a shame. I am looking at alternatives.

Reply:
Maybe fixing it in lightning is not a bad idea, but it we'll have to wait for a release.

Reply:
We can keep the workaround here until it's fixed in lightning.